### ISSUE TITLE
panic: runtime error: slice bounds out of range [:10] with capacity 0 goroutine 70

### ISSUE THREAD

**Issue by d-Pixie** [zinclabs/zinc#163]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Which service(s) is this relate to?**
Stateful set in Kubernetes

**Are you currently working around this issue?**
Not that I know. Tried restarting the service, but it gets into an immediate crash loop.

**Additional context**
Running through helm, latest chart. Ingesting via fluend-bit on GKE.
It worked for a little while, but within 10 minutes or so the Zinc pod crashed with the message below and keeps crashing on restart.

**Attachments**
```
panic: runtime error: slice bounds out of range [:10] with capacity 0 goroutine 70
[running]:
github.com/blugelabs/ice.(*Segment).getDocStoredOffsets(0xc00032ebb0, 0x0) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/read.go:52 +0x266
github.com/blugelabs/ice.(*Segment).getDocStoredMetaAndUnCompressed(0xc00032ebb0, 0xc0002ecc00) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/read.go:22 +0x25
github.com/blugelabs/ice.(*Segment).visitDocument(0xc00032ebb0, 0xc0002ecc00, 0x40d085, 0xc00072f588) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/segment.go:197 +0x54
github.com/blugelabs/ice.mergeStoredAndRemapSegment(0xc00032ebb0, 0x0, {0xc00012f2c0, 0xc, 0x203000}, 0x203000, 0xc00072f748, {0x0, 0x0, 0x0}, ...) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/merge.go:728 +0x214
github.com/blugelabs/ice.mergeStoredAndRemap({0xc0002ac910, 0xa, 0x39}, {0xc0002ac870, 0xa, 0xc0002d58f8}, 0x40d085, {0xc00000a5a0, 0x1e, 0x1e}, ...) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/merge.go:675 +0x57e
github.com/blugelabs/ice.mergeToWriter({0xc0002ac910, 0xa, 0xa}, {0xc0002ac870, 0xa, 0xa}, 0x2ac910, 0xc000277040, 0xc0002d59f0) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/merge.go:131 +0x19f
github.com/blugelabs/ice.mergeSegmentBasesWriter({0xc0002ac910, 0xa, 0xa}, {0xc0002ac870, 0xa, 0xa}, {0x1f5a160, 0xc0002ecbc0}, 0x401, 0xc0002e8de0) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/merge.go:97 +0xfd
github.com/blugelabs/ice.merge({0xc000451360, 0xa, 0x9ae5f1}, {0xc0002ac870, 0xa, 0xa}, {0x1f5a160, 0xc0002ecbc0}, 0xc000310c08) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/merge.go:86 +0xba
github.com/blugelabs/ice.(*Merger).WriteTo(0xc0002ac8c0, {0x1f5d480, 0xc000310c08}, 0x117aacd) /go/pkg/mod/github.com/zinclabs/ice@v0.2.1-0.20220504041539-ace50a7ef71f/merge.go:49 +0x14d
github.com/blugelabs/bluge/index.(*FileSystemDirectory).Persist(0xc00024c400, {0x117aacd, 0x4}, 0xc0002ac870, {0x7f498917f098, 0xc0002ac8c0}, 0x100000) /go/pkg/mod/github.com/blugelabs/bluge@v0.1.9/index/directory_fs.go:125 +0x1c6
github.com/blugelabs/bluge/index.(*Writer).merge(0xc000199680, {0xc000451360, 0x0, 0x2}, {0xc0002ac870, 0x2, 0xc0002ec500}, 0x2817968) /go/pkg/mod/github.com/blugelabs/bluge@v0.1.9/index/merge.go:368 +0xb3
github.com/blugelabs/bluge/index.(*Writer).executeMergeTask(0xc000199680, 0xc00007c960, 0xc0002db8c0) /go/pkg/mod/github.com/blugelabs/bluge@v0.1.9/index/merge.go:144 +0x125
github.com/blugelabs/bluge/index.(*Writer).planMergeAtSnapshot(0xc000199680, 0x0, 0xc0000dab00, {0xa, 0x4c4b40, 0x4024000000000000, 0xa, 0x7d0, 0x4000000000000000, 0x0, ...}) /go/pkg/mod/github.com/blugelabs/bluge@v0.1.9/index/merge.go:118 +0x305
github.com/blugelabs/bluge/index.(*Writer).mergerLoop(0xc000199680, 0x0, 0x0) /go/pkg/mod/github.com/blugelabs/bluge@v0.1.9/index/merge.go:56 +0x25f created by
github.com/blugelabs/bluge/index.OpenWriter /go/pkg/mod/github.com/blugelabs/bluge@v0.1.9/index/writer.go:131 +0x85d
```

**Comment by hengfeiyang** [comment]
Have beed fixed, we will release new version in next week, or you can run our test version fixed:

```
public.ecr.aws/zinclabs/zinc:test
```

**Comment by d-Pixie** [comment]
@hengfeiyang great. No worries, we'll wait for the new version.

Fantastic tool, appreciate all of your work.

**Comment by hengfeiyang** [comment]
@d-Pixie It was fixed in version v0.2.3.
